main.go

```diff
@@ -64,68 +50,64 @@ func applyVirtualMachineExport(vmNamespace, vmName string) error {
 		},
 	}
 
-	_, err = client.VirtualMachineExport(vmNamespace).Create(context.Background(), vmExport, metav1.CreateOptions{})
+	_, err := client.VirtualMachineExport(vmNamespace).Create(context.Background(), vmExport, metav1.CreateOptions{})
 	return err
 }
 
-func downloadVirtualMachineDiskImage(vmName, volumeName string) error {
-	log.Printf("Downloading disk image from %s Virtual Machine...\n", vmName)
-
-	cmd := exec.Command("usr/bin/virtctl", "vmexport", "download", vmName, "--vm", vmName, "--volume", volumeName, "--output", diskPath)
-	cmd.Stdout = os.Stdout
-	cmd.Stderr = os.Stderr
+func getRawDiskUrlFromVirtualMachineExport(client kubecli.KubevirtClient, vmNamespace, vmName string) (string, error) {
+	log.Println("Waiting for VirtualMachineExport to be ready...")
 
-	if err := cmd.Run(); err != nil {
-		return err
+	vmExport, err := getVirtualMachineExportOnceReady(client, vmNamespace, vmName)
+	if err != nil {
+		return "", err
 	}
 
-	if fileInfo, err := os.Stat(diskPath); err != nil || fileInfo.Size() == 0 {
-		return fmt.Errorf("File does not exist or is empty.")
+	if vmExport.Status.Links == nil {
+		return "", fmt.Errorf("No links found in VirtualMachineExport status.")
 	}
 
-	log.Println("Download completed successfully.")
-	return nil
-}
-
-func decompressVirtualMachineDiskImage() error {
-	log.Println("Decompressing downloaded disk image...")
-
-	diskImg, err := os.Open(diskPath)
-	if err != nil {
-		return err
+	if vmExport.Status.Links.Internal == nil {
+		return "", fmt.Errorf("No internal links found in VirtualMachineExport status.")
 	}
-	defer diskImg.Close()
 
-	gzipReader, err := gzip.NewReader(diskImg)
-	if err != nil {
-		return err
+	for _, volume := range vmExport.Status.Links.Internal.Volumes {
+		for _, format := range volume.Formats {
```

Comment:
Would be possible to add check if the volume name is the same as defined in task parameter?

Reply:
Yes good catch. I'll add this check.